vertx-sql-client/src/main/java/io/vertx/sqlclient/impl/PoolImpl.java

```diff
@@ -155,6 +158,44 @@ public Future<SqlConnection> getConnection() {
     });
   }
 
+  public <T> Future<@Nullable T> withPropagatedTransaction(Function<SqlConnection, Future<@Nullable T>> function) {
+    Context context = Vertx.currentContext();
+    SqlConnection sqlConnection = context.getLocal(PROPAGATABLE_CONNECTION);
+    if (sqlConnection == null) {
+      return initializePropagatedConnectionAndTransaction(function);
+    }
+    return Future.succeededFuture(sqlConnection)
```

Comment:
instead here we need to use `context.succeededFuture` (context must be casted to `ContextInternal`)

Reply:
Done. However, the authorship got messed up as I did this from another machine (both authorships should refer to the same person, me). Do you need me to sign the ECA once again?

Reply:
I tried to register an Eclipse user for my work email and signing the ECA there as well. Something changed on the verification page and it says that it's signed, but it still shows as not valid